File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/models/api/fields.ts
import { SingleTextField } from './singletextfield.model';
import { MultiTextField } from './multitextfield.model';
import { SingleNumberField } from './singlenumberfield.model';
import { MultiNumberField } from './multinumberfield.model';
import { SingleBooleanField } from './singlebooleanfield';
import { SingleDecimalField } from './singledecimalfield.model';
import { MultiDecimalField } from './multidecimalfield.model';
import { SingleDateField } from './singledatefield.model';
import { MultiDateField } from './multidatefield.model';
import { SingleListField } from './singlelistfield.model';
import { SingleRelationshipField } from './singlerelationshipfield.model';
import { MultiRelationshipField } from './multirelationshipfield.model';
import { SingleRelationshipListField } from './singlerelationshiplistfield.model';
import { SingleHyperlinkField } from './singlehyperlinkfield.model';
import { MultiHyperlinkField } from './multihyperlinkfield.model';

export class Fields {
	public constructor(
		public singleTextFields: SingleTextField[] = [],
		public multiTextFields: MultiTextField[] = [],
		public singleNumberFields: SingleNumberField[] = [],
		public multiNumberFields: MultiNumberField[] = [],
		public singleBooleanFields: SingleBooleanField[] = [],
		public singleDecimalFields: SingleDecimalField[] = [],
		public multiDecimalFields: MultiDecimalField[] = [],
		public singleDateFields: SingleDateField[] = [],
		public multiDateFields: MultiDateField[] = [],
		public singleListFields: SingleListField[] = [],
		public singleRelationshipFields: SingleRelationshipField[] = [],
		public multiRelationshipFields: MultiRelationshipField[] = [],
		public singleRelationshipListFields: SingleRelationshipListField[] = [],
		public singleHyperlinkFields: SingleHyperlinkField[] = [],
		public multiHyperlinkFields: MultiHyperlinkField[] = []
	) {}
}

File: Mavim-services/Topic/Mavim.Manager.Api.Topic/v1/Models/Fields.cs
using System.Collections.Generic;
using Service = Mavim.Manager.Api.Topic.Services.v1.Models.Fields;

namespace Mavim.Manager.Api.Topic.v1.Models
{
    /// <summary>
    /// Fields
    /// </summary>
    public class Fields
    {
        /// <summary>
        /// SingleTextFields
        /// </summary>
        public List<Service.SingleTextField> SingleTextFields { get; set; }
        /// <summary>
        /// MultiTextFields
        /// </summary>
        public List<Service.MultiTextField> MultiTextFields { get; set; }
        /// <summary>
        /// SingleNumberFields
        /// </summary>
        public List<Service.SingleNumberField> SingleNumberFields { get; set; }
        /// <summary>
        /// MultiNumberFields
        /// </summary>
        public List<Service.MultiNumberField> MultiNumberFields { get; set; }
        /// <summary>
        /// SingleBooleanFields
        /// </summary>
        public List<Service.SingleBooleanField> SingleBooleanFields { get; set; }
        /// <summary>
        /// SingleDecimalFields
        /// </summary>
        public List<Service.SingleDecimalField> SingleDecimalFields { get; set; }
        /// <summary>
        /// MultiDecimalFields
        /// </summary>
        public List<Service.MultiDecimalField> MultiDecimalFields { get; set; }
        /// <summary>
        /// SingleDateFields
        /// </summary>
        public List<Service.SingleDateField> SingleDateFields { get; set; }
        /// <summary>
        /// MultiDateFields
        /// </summary>
        public List<Service.MultiDateField> MultiDateFields { get; set; }
        /// <summary>
        /// SingleListFields
        /// </summary>
        public List<Service.SingleListField> SingleListFields { get; set; }
        /// <summary>
        /// SingleRelationshipFields
        /// </summary>
        public List<RelationshipField> SingleRelationshipFields { get; set; }
        /// <summary>
        /// MultiRelationshipFields
        /// </summary>
        public List<MultiRelationshipField> MultiRelationshipFields { get; set; }
        /// <summary>
        /// SingleRelationshipListFields
        /// </summary>
        public List<RelationshipListField> SingleRelationshipListFields { get; set; }
        /// <summary>
        /// SingleHyperlinkFields
        /// </summary>
        public List<Service.SingleHyperlinkField> SingleHyperlinkFields { get; set; }
        /// <summary>
        /// MultiHyperlinkFields
        /// </summary>
        public List<Service.MultiHyperlinkField> MultiHyperlinkFields { get; set; }
    }
}